### ISSUE TITLE
No documentation available for chartjs-plugin-annotation

### ISSUE THREAD

**Issue by RoyStout** [chartjs/chartjs-plugin-annotation#217]
Why can I not find any documentation  on this library?
Or where is it hidden?
Searches take me to chart,js, which has a link to this github, but still no documentation.
Are we supposed to guess at the syntax?
Are we supposed to guess at the possible parameters and their possible values?
No documentation IS an issue.

**Comment by strixy** [comment]
tldr; 
Docs for "this library" (annotation plugin for chart.js) https://github.com/chartjs/chartjs-plugin-annotation and scroll down a bit.
Docs for Chart.JS (required for working with this library) https://www.chartjs.org/docs/latest/ 

It's not clear to me which library's documentation it is that you're looking for so I'll include both of the possibilities that I think you might be looking for.

It's common practice on GitHub to keep documentation in a file called readme.md. That file can always be found in the root of the source. Hold on though! No one expects you to know that.

You'll find the contents of that readme.md are included (rendered) on the main page of the repository. That's also the default page you likely found earlier. That's also the page that has the source code file navigator on it. You can get there from here by clicking on the "<> Code" tab above. Hold on! Don't do that yet.

**You'll need to scroll down below the code navigator**. 

Welcome to GitHub.

**Comment by paulhayessilex** [comment]
> tldr;
> Docs for "this library" (annotation plugin for chart.js) https://github.com/chartjs/chartjs-plugin-annotation and scroll down a bit.
> Docs for Chart.JS (required for working with this library) https://www.chartjs.org/docs/latest/
> 
> It's not clear to me which library's documentation it is that you're looking for so I'll include both of the possibilities that I think you might be looking for.
> 
> It's common practice on GitHub to keep documentation in a file called readme.md. That file can always be found in the root of the source. Hold on though! No one expects you to know that.
> 
> You'll find the contents of that readme.md are included (rendered) on the main page of the repository. That's also the default page you likely found earlier. That's also the page that has the source code file navigator on it. You can get there from here by clicking on the "<> Code" tab above. Hold on! Don't do that yet.
> 
> **You'll need to scroll down below the code navigator**.
> 
> Welcome to GitHub.

Fair enough, but I can't get it working with React for the life of me, can you provide React examples for the chartjs v3.0.0-beta.7

**Comment by strixy** [comment]
tldr1; You most likely have a race condition. Try using this.render() after you resolve your promise or async/await.

tldr2; The Issues tab on GitHub is where we developers, the consumers of the package, give support to the maintainers of a specific package. StackOverflow is where we give (and receive) support for (and from) the larger community where the integration and use of those packages is concerned.

I can see you have an issue, but it appears to me to be more related to your individual use of the code as opposed to a bug in the code itself. By all means, correct me if I'm wrong on that assumption. If I am wrong can you please include more details on any specific error messages you may be seeing? I would need more details to be able to give a more detailed answer. (This is the most common way we can help support the maintainers of any package).

However, based on what you've said so far the only thing I can think of that might assist you is to tell you that Chart JS is not reactive (it is event driven) nor is it's configuration object observable. You're probably already aware that you have to call `this.Chart.render()`, but you also have to make absolutely sure your config and data are set **prior** to doing so. If you call `render()` before your data is returned from the server and before it's set in the config object (which is a deep object structure in and of itself and not directly observable anyway) you will have a race condition. In other words, you may be rendering the chart before the data is available to even be rendered and Chart JS does not react to those changes in your data model. Try using `this.Chart.render()` after you have resolved your promise, in `.then()` or after async/await. I don't know your specific use case as you haven't shared any examples of your code.

I've long mused that "Issues" is an inappropriate label for this section, but that's perhaps an example of an issue in and of itself. This is more so the place to report bugs (aka issues) relevant to the Chart JS code base. When someone has questions about the use of the code (eg "how do I..."), it's best to search StackOverflow and, failing any acceptable results, it's best to ask your question there. The Chart JS community from here also helps out there, but more importantly, React community members are there as well. Some subset of those two communities will also work with Chart JS and are probably more familiar with your specific situation than I am. I haven't touched React since switching to Vue 4+ years ago and even then I didn't integrate Chart JS.

The answer to "why" is that the "issues" here are tracked and associated to code changes. The "Issues" tab is meant to assist the developers track their workflow. The purpose of this section of GitHub is to give the community a way to support the developers of the package in question. We support them here primarily by sharing the steps to reproduce a bug in the software so they can fix it faster. I know what you might be thinking, why not offer support for the software they're creating and hosting on GitHub right here on GitHub? To which I will point to your own question above as an example. Your original concern involved the confluence of two unique packages; React and Chart JS. You can't really ask for help with Chart JS on the React JS GitHub page anymore than you might receive help here for your React question. I'll bet your first post is related to that conundrum. So then, where does one go? There needs to be a way for you to get help from a broader community with experience in both? That's where StackOverflow comes in. There are other sites out there besides Stack and many open source projects have dedicated forums. In this case, I would recommend checking and potentially posting on Stack.

Consider answering (at least) 2 questions on Stack for every 1 question you ask. If you aren't sure yet how to do all that or don't feel comfortable answering any questions yet, don't worry about it. Come back around to it later when you do. Karma is patient. Cheers.

**Comment by etimberg** [comment]
There is a react example in https://github.com/chartjs/chartjs-plugin-annotation/issues/216#issuecomment-754511081
